mkosi

```diff
@@ -268,59 +271,134 @@ def prepare_esp(args, loopdev):
 
     print_step("Formatting ESP partition completed.");
 
-def mkfs_ext4(label, mount, loopdev, partno):
-    subprocess.run(["mkfs.ext4", "-L", label, "-M", mount, partition(loopdev, partno)], check=True)
+def mkfs_ext4(label, mount, dev):
+    subprocess.run(["mkfs.ext4", "-L", label, "-M", mount, dev], check=True)
+
+def prepare_luks(args):
+
+    print_step("LUKS setup")
+    passphrase = luks_read_passphrase()
+
+    keys = {}
+    if args.luks_encryption == "all":
+        keys["root"] = passphrase
+        keys["home"] = passphrase
+        keys["srv"] = passphrase
+    elif args.luks_encryption == "data":
```

Comment:
maybe make luksencryption an `Enum`, in the style of the existing `Distribution` and `OutputFormat` enums?